crates/wasmtime/src/trap.rs

```diff
@@ -121,16 +126,22 @@ pub(crate) fn from_runtime_box(
         }
         wasmtime_runtime::TrapReason::Wasm(trap_code) => (trap_code.into(), None),
     };
-    match backtrace {
-        Some(bt) => {
+
+    match (coredumpstack, backtrace) {
+        (None, None) => error,
+        (None, Some(bt)) => {
             let bt = WasmBacktrace::from_captured(store, bt, pc);
-            if bt.wasm_trace.is_empty() {
-                error
-            } else {
+            if !bt.wasm_trace.is_empty() {
                 error.context(bt)
+            } else {
+                error
             }
         }
-        None => error,
+        (Some(coredump), _) => {
+            let bt = WasmBacktrace::from_captured(store, coredump.bt, pc);
+            let cd = WasmCoreDump::new(store, bt);
+            error.context(cd)
+        }
```

Comment:
I think if we have a backtrace and a coredump, we want to attach them both as context to the error, so I think it would be better to do something like this:

```rust
let mut error = ...;

if let Some(bt) = backtrace {
    let bt = WasmBacktrace::from_captured(store, bt, pc);
    if !bt.wasm_trace.is_empty() {
        error = error.context(bt);
    }
}

if let Some(cd) = coredump {
    let bt = WasmBacktrace::from_captured(store, cd.bt, pc);
    let cd = WasmCoreDump::new(store, bt);
    error = error.context(cd);
}

error
```